### PR TITLE
ci(core): fix node-dev Build.ts linting

### PR DESCRIPTION
**Pull request by OlegIvaniv**
Use the new parseJson helper for parsing of `tsconfig-build.json` inside `node-dev/src/Build.ts`

### CODE REVIEW

**Review by ivov — CHANGES_REQUESTED** [review]
Can we please fix using the helper from #4403?

**Comment by netroy** [comment]
@ivov any idea why the issue wasn't caught in https://github.com/n8n-io/n8n/pull/4403 ?
Is it an issue with the diff plugin?

File: packages/node-dev/src/Build.ts [hunk 1]
```diff
@@ -22,12 +23,12 @@ export async function createCustomTsconfig() {
 	// Read the tsconfig file
 	const tsConfigString = await readFile(tsconfigPath, { encoding: 'utf8' });
 	// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-	const tsConfig = JSON.parse(tsConfigString);
+	const tsConfig = jsonParse<IDataObject>(tsConfigString);
```
Comment (ivov) [inline packages/node-dev/src/Build.ts:26]:
```suggestion
	const tsConfig = jsonParse<{ include: string[] }>(tsConfigString);
```

File: packages/node-dev/src/Build.ts [hunk 2]
```diff
@@ -22,12 +23,12 @@ export async function createCustomTsconfig() {
...
 
 	// Set absolute include paths
 	const newIncludeFiles = [];
 	// eslint-disable-next-line no-restricted-syntax
-	for (const includeFile of tsConfig.include) {
+	for (const includeFile of tsConfig.include as string[]) {
```
Comment (ivov) [inline packages/node-dev/src/Build.ts:31]:
```suggestion
	for (const includeFile of tsConfig.include) {
```

**Comment by ivov** [comment]
> @ivov any idea why the issue wasn't caught in #4403 ? Is it an issue with the diff plugin?

Exactly, when we add a new rule or increase severity to `error`, we have to run a full lint. I'll add a tech debt story to check and enforce this.

**Review by ivov — APPROVED** [review]

**Comment by janober** [comment]
Got released with `n8n@0.200.0`

**Pull request merged**